feat(vertex): support 'trafficType' in provider usageMetadata

danielamitay opened the pull request:
## Background
- When using google vertex, knowing whether or not a request was handled by provisioned throughput or the on demand shared pool is useful for both billing/cost measurement purposes as well as analytics/data segregation purposes.
- Currently this isn't exposed/available via the ai-sdk, but it is available under the hood.

## Summary
- Vertex returns a field in the usage metadata in the response indicating [whether the request was handled by the on-demand pool, or by provisioned throughput](https://cloud.google.com/vertex-ai/generative-ai/docs/reference/rest/v1/GenerateContentResponse#TrafficType). This is useful for segregating analytics. To surface it to end users of ai-sdk we need to at least add it to the `usageSchema`

<img width="360" height="288" alt="Screenshot 2025-11-11 at 10 33 28 AM" src="https://github.com/user-attachments/assets/59d6fde5-1e21-4c76-b537-f10082da3987" />

## Manual Verification
- Updated google-vertex.ts script runs and shows:
```
Token usage: {
  inputTokens: 19,
  outputTokens: 605,
  totalTokens: 1817,
  reasoningTokens: 1193,
  cachedInputTokens: undefined
}
Finish reason: stop
Traffic type: ON_DEMAND
```

## Checklist
- [ ] Tests have been added / updated (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] A _patch_ changeset for relevant packages has been added (for bug fixes / features - run `pnpm changeset` in the project root)
- [x] I have reviewed this pull request (self-review)

danielamitay committed:
Update google usageSchema to support passing through "trafficType" from Vertex requests

danielamitay committed:
Add changeset

danielamitay committed:
Merge branch 'main' into support-vertex-trafficType

danielamitay committed:
Merge branch 'main' into support-vertex-trafficType

gr2m reviewed (commented):
updating the schema is not sufficient. Please follow our https://github.com/vercel/ai/edit/main/.github/pull_request_template.md and make sure you test your changes end-to-end, e.g. using [examples/ai-core/src/stream-text/google.ts](https://github.com/vercel/ai/blob/7dea60ec21c7a95dab8c91666ca7d5e1ec96e584/examples/ai-core/src/stream-text/google.ts)

danielamitay committed:
Update google-vertex.ts

danielamitay committed:
Merge branch 'main' into support-vertex-trafficType

danielamitay commented:
@gr2m apologies for this PR also not following the PR template. Fixed and I won't do that again. I updated the google-vertex.ts example. While this change is in the google package, the response payload only includes this value when served by Vertex. The vertex request example was also updated from Gemini 1.5 Pro to 2.5 Pro because the script failed otherwise as the 1.5 Pro model is no longer served. Here is the truncated output from running that script:

```
Alright, settle down, settle down. I've got a new holiday for you, and it's going to change your life... eventually.

[...truncated...]

The official greeting of the holiday isn't "Happy Shoulda-Coulda-Woulda Day!" It's just making eye contact with someone across the room, giving a slow, weary nod, and saying, **"Eh, maybe tomorrow."**

The best part? If you completely forget about the holiday until the next day, you're not late. You're celebrating it perfectly.
Token usage: {
  inputTokens: 19,
  outputTokens: 605,
  totalTokens: 1817,
  reasoningTokens: 1193,
  cachedInputTokens: undefined
}
Finish reason: stop
Traffic type: ON_DEMAND
```

danielamitay committed:
prettier-fix

danielamitay committed:
Merge branch 'support-vertex-trafficType' of https://github.com/danielamitay/ai into support-vertex-trafficType

gr2m committed:
log out all of `usageMetadata` in example

gr2m reviewed (approved):
Thanks Daniel! I tested it locally and verified it works as expected.
